util/annotations/helpers.go

```diff
@@ -55,6 +55,7 @@ func AddAnnotations(o metav1.Object, desired map[string]string) bool {
 	annotations := o.GetAnnotations()
 	if annotations == nil {
 		annotations = make(map[string]string)
+		o.SetAnnotations(annotations)
```

Comment:
THis should probably be set on line 65 only if the annotations have changed. We probably want to avoid turning a nil map into an instance if there are no changes?

Reply:
Also let's add a test to that effect?

Reply:
I actually don't think we need this at all, maps are references in Go so modifying `annotations` will also modify the object annotation

Reply:
The problem is that we're just creating a new map and this doesn't affect the object we passed in. The added test case below fails without the code change here.

Reply:
@vincepri I could do it either here with the additional condition `len(desired) > 0` or below. Below I would also have to add some kind of check or do it for every annotation which we want to add. Both are fine for me.

I'll add a test so we don't change the nil to an empty map if we don't pass in annotations

Reply:
Ah I'll just add the following at the top
```golang
if len(desired) == 0 {
  return false
}
```

Reply:
If we do set the annotations at the bottom, I think it should pass?

Reply:
> The problem is that we're just creating a new map and this doesn't affect the object we passed in. The added test case below fails without the code change here.

oh I see now, you are right.

Reply:
I added the guard clause above.
So the logic is like:
* if we don't need to do anything +> return false
* if the map is nil (and we know have to do something when len(desired)>0) => create and set annotation

But I can move it around if you prefer it another way.